feelpp/feel/feelmesh/convex.hpp

```diff
@@ -106,5 +106,10 @@ class Convex : public ConvexBase
 private:
 
 };
+
+template<typename Convex>
+struct ReduceDim {
```

Comment:
I think, it should be move inside simplex/hypercurbe class, and maybe add a static check about a null dim.

But do you know that you have the type `trace_mesh_type` in Mesh. I think it's exactly what you want,no? (mesh on topological face) If yes, you can remove these additions.
Moreover, you can write `Mesh::trace_mesh_type::trace_mesh_type` in 3d for a mesh defined on edges

Reply:
Indeed, I didn't know about `trace_mesh_type`, I'll revert these changes and use that. Thanks